### ISSUE TITLE
Investigate getting $using: working in Pages/Layouts/Elements

### ISSUE THREAD

**Issue by Badgerati** [Badgerati/Pode.Web#138]
Pode itself has support for `$using:` variables in Routes, Middleware, Auth, Timers, etc, etc.

It would be useful if Pode.Web's Pages, Elements, and Layouts could do the same. It'll be a little trickier since they're wrappers around Routes, but worth investigating.

**Comment by Badgerati** [comment]
This should also cover `$state:`, `$session:`, and `$secret:`.

**Comment by phdavis** [comment]
Would be super handy to get. 

We've had Pode running for quite some time and heavily use $using in our routes and such. This confused me for a bit while we were testing Web. Get/Set-PodeState does seem to work fine, but trying to keep code bases similar (i.e. an API repo for Pode and a WEB repo for Pode.Web) it would be nice to have the same code to prevent confusion.

We'd either need $using working to do this, or we'll want to go back and refactor our API code base to use the Get/Set functions.

**Comment by Badgerati** [comment]
Agreed, it can be confusing; the reason it doesn't work is because Pages created via Pode.Web, even though they create Routes behind the scenes, suffer from ScriptBlock scoping issues - so `$using:` can't see the variables correctly.

I'd either have to re-implement the logic again in Pode.Web, or find a way to make the logic re-usable some how from Pode - the latter I'm going to try and figure out, as it would help other extension modules in the future if it's doable.
